test/applications/test_clique.py

```diff
@@ -0,0 +1,73 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2018, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+""" Test Clique class"""
+
+from test.optimization_test_case import QiskitOptimizationTestCase
+import networkx as nx
+
+from qiskit_optimization import QuadraticProgram
+from qiskit_optimization.algorithms import (OptimizationResult,
+                                            OptimizationResultStatus)
+from qiskit_optimization.applications.clique import Clique
+from qiskit_optimization.problems import (Constraint, QuadraticObjective, VarType)
+
+
+class TestClique(QiskitOptimizationTestCase):
+    """ Test Clique class"""
+
+    def setUp(self):
+        """Set up for the tests"""
+        super().setUp()
+        self.graph = nx.gnm_random_graph(5, 8, 123)
+        op = QuadraticProgram()
+        for _ in range(5):
+            op.binary_var()
+        self.result = OptimizationResult(
+            x=[1, 0, 1, 1, 1], fval=4, variables=op.variables,
+            status=OptimizationResultStatus.SUCCESS)
+
+    def test_to_quadratic_program(self):
+        """Test to_quadratic_program"""
+        clique = Clique(self.graph)
+        op = clique.to_quadratic_program()
```

Comment:
I guess we have any equality method on Quadratic Program such that some reference QP could be generated here and then one simple test for equality made? I see there are similar logic in other tests  to test the QP is as expected. Maybe some for of equals would be useful? If not in QP itself then maybe as some util function, perhaps in the tests here even, that would help avoid doing this test explicitly piece by piece each time we need it.

Reply:
No, there is no equality method of `QuadraticProgram`. It is generally used only for unit tests. So, I recommend to add an equality function to a unit test.

Reply:
Somehow `QuadraticProgram` does not have a method for equality. It test explicitly even in the test of `QuadraticProgram`. I didn't think about it, but, you're right. I could make some helper function like `assertQPEqual` for those tests. I will use a helper function next time as you taught me.